src/aggregate/aggregate_exec.c

```diff
@@ -685,7 +685,6 @@ static int buildRequest(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
   // ctx is always assigned after ApplyContext
   if (rc != REDISMODULE_OK) {
     RS_LOG_ASSERT(QueryError_HasError(status), "Query has error");
-    goto done;
```

Comment:
and what if additional steps will be added in the future? why do we care to remove this `goto`?

Reply:
Then this will need to be inserted. At the moment it's not doing anything so I see no reason for it to be here. If you particularly like it I can leave it.